cloud-controller-manager/do/loadbalancers.go

```diff
@@ -780,6 +785,30 @@ func getHostname(service *v1.Service) string {
 	return strings.ToLower(service.Annotations[annDOHostname])
 }
 
+// healthCheckPort returns the health check port specified, defaulting
+// to the first port in the service otherwise.
+func healthCheckPort(service *v1.Service) (int, error) {
+	ports, err := getPorts(service, annDOHealthCheckPort)
+	if err != nil {
+		return -1, fmt.Errorf("failed to get health check port: %v", err)
```

Comment:
It might be more idiomatic to return the zero value for the non-error parameter, i.e., 0 instead of -1.

(Similar for other error cases below.)